Update dependencies

LouisDISPA opened the pull request:
Hello, thank you for this amazing work. It works on my machine with only a small issue. (the color profile for thumbnails)
I quickly updated the dependencies to see if this would fix it but sadly it did not.

Here is a PR with the updates, you are free to do whatever you want with it.
If you have any request of modification feel free to ask. I have some spare time.

saschanaz commented on README.md:
```diff
@@ -9,9 +9,11 @@ A JPEG XL (*.jxl) WIC decoder to render thumbnails on Windows File Explorer or v
 1. Move to your download directory
 1. `regsvr32 jxl_winthumb.dll`, or to uninstall, `regsvr32 /u jxl_winthumb.dll`.
 
+You might need to restart windows for the extention to work.
```
Restarting `explorer.exe` should be enough.

saschanaz replied:
```suggestion
You might need to restart `explorer.exe` for the component to work.
```

("Component" because C of both WIC and COM is Component.)

saschanaz reviewed (commented):
Wow, thank you for the chore! Yeah, I think fixing the issue needs some more investigation (I think the issue might just be about emitting the 128bit float pixel because I guess that's a rare use case. I wonder the new jxl-oxide supports lower precision number types?

Some nits:

saschanaz commented on tests/wic.rs:
```diff
@@ -5,7 +5,7 @@ use windows::Win32::UI::Shell::SHCreateMemStream;
 #[test]
 fn basic() {
-    unsafe { CoInitialize(None) }.expect("CoInitialize");
+    unsafe { CoInitialize(None) }.unwrap();
```
Why this change? 🤔

LouisDISPA replied:
CoInitialize now return a custom type that only have `unwrap` and not `expect`. But I found a method to change it to a standard `Result` type.

LouisDISPA commented:
Welcome, I fixed the nits!  
I also added the cmd to restart `explorer.exe`.

> I think fixing the issue needs some more investigation (I think the issue might just be about emitting the 128bit float pixel because I guess that's a rare use case. I wonder the new jxl-oxide supports lower precision number types?

Good to know, I don't have a lot of experience on image processing but I might try to have a look.

LouisDISPA commented:
I pushed a last commit that remove some unwrap by using pattern matching. The logic is the same but if you prefer the old code I can revert it.

goodusername123 commented on Cargo.toml:
```diff
@@ -10,13 +10,13 @@ crate-type = ["rlib", "cdylib"]
+simple-logging = "2.0"
+log = "0.4"
+winreg = "0.52.0"
+jxl-oxide = "0.7"
```
Version [0.7.1](https://github.com/tirr-c/jxl-oxide/releases/tag/0.7.1) of jxl-oxide released a few hours ago.

saschanaz commented:
> I pushed a last commit that remove some unwrap by using pattern matching. The logic is the same but if you prefer the old code I can revert it.

I do like them but diffing is now harder, can we do that in a separate PR?

saschanaz commented on Cargo.toml:
```diff
@@ -10,13 +10,13 @@ crate-type = ["rlib", "cdylib"]
-winreg = "0.10.1"
-jxl-oxide = "0.5.1"
+simple-logging = "2.0"
+log = "0.4"
```
Was this done by some tool or by hand? Not sure the point to remove the patch versions.

LouisDISPA replied:
This was done by hand. Since the logging crates are stable I removed the patch versions. Maybe jxl-oxide should have it just to be sure. As you wish!

saschanaz commented on README.md:
```diff
@@ -9,9 +9,12 @@ A JPEG XL (*.jxl) WIC decoder to render thumbnails on Windows File Explorer or v
 1. `regsvr32 jxl_winthumb.dll`, or to uninstall, `regsvr32 /u jxl_winthumb.dll`.
 
+You might need to restart `explorer.exe` for the component to work.  
+You can use this command: `taskkill /f /im explorer.exe & start explorer.exe`
```
```suggestion
You can use this command: `taskkill /f /im explorer.exe && start explorer.exe`
```

`&` does different thing in pwsh, while it seems both work in cmd.

saschanaz commented on src/properties.rs:
```diff
@@ -1,21 +1,21 @@
 use std::io::BufReader;
 
+use crate::winstream::WinStream;
```
I kinda tried to put all internal imports after external imports although there was an exception. Meaning, I'd be happier to keep WinStream after windows imports.

LouisDISPA replied:
Sorry it's the cargo fmt command that moved everything.

saschanaz commented on src/dll.rs:
```diff
@@ -113,7 +113,7 @@ pub extern "stdcall" fn DllMain(
     rclsid: *const GUID,
     riid: *const GUID,
-    pout: *mut *const core::ffi::c_void,
+    pout: *mut *mut core::ffi::c_void,
```
Oh, unknown.query now accepts mut mut. Seems it also accepts `*const GUID` instead of a reference now, can you remove `&*` there while we are at it?

LouisDISPA replied:
Good catch!

saschanaz reviewed (commented):
Looking good! A few more nits:

LouisDISPA commented:
> Looking good! A few more nits:
Done!

saschanaz commented on README.md:
```diff
@@ -9,9 +9,12 @@ A JPEG XL (*.jxl) WIC decoder to render thumbnails on Windows File Explorer or v
 1. Move to your download directory
 1. `regsvr32 jxl_winthumb.dll`, or to uninstall, `regsvr32 /u jxl_winthumb.dll`.
 
+You might need to restart `explorer.exe` for the component to work.  
+You can use this command: `taskkill /f /im explorer.exe && start explorer.exe`
```
```suggestion
You might need to restart `explorer.exe` or any programs that use the dll before updating it. Get the list of such programs using `tasklist /m jxl_winthumb.dll` and kill them e.g. with `taskkill /f /im explorer.exe && start explorer.exe`.
```

saschanaz commented:
Thank you so much! So helpful 👍🙏
